grunttasks/generate-valid-styles.js

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var path = require('path');
+var stripSpecialChars = require('./strip-special-chars');
+var VALID_STYLES_FILE_NAME = 'validStyles.json';
+
+module.exports = function (grunt) {
+  'use strict';
+   grunt.registerMultiTask('generate-valid-styles', 'Generate the list of valid styles for translation', function () {
+    var validStylesRegex = /style=('|")[\s\S]*?('|")/ig;
+    var validStylesArray = [];
+
+    // iterate through .pot file and extract the styles
+    this.files[0].src.forEach(function (src) {
+      var contents = grunt.file.read(src);
+      if (validStylesRegex.test(contents)) {
+        validStylesArray = validStylesArray.concat(contents.match(validStylesRegex));
+      }
+    });
+    validStylesArray = validStylesArray.map(stripSpecialChars);
+    // write out the styles to a temporary JSON file
+    grunt.file.write(path.join(this.files[0].dest, VALID_STYLES_FILE_NAME), JSON.stringify(validStylesArray));
```

Comment:
What is the full path of the json file? Does this add the temporary json file in with the rest of the .po/.pot files? Temporary files should not pollute the original directory tree, let's use a subdirectory that can be cleaned for this purpose. In the content server, we use `.tmp`